### ISSUE TITLE
New feature: deep copy helper function for elements

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/spatialdata#287]
Sometimes we need to deep copy elements (=spatial elements or the `AnnData` table), but `copy.deepcopy()` can lead to subtle problems:
https://github.com/scverse/spatialdata/issues/286
https://github.com/scverse/spatialdata/issues/269

So probably we want to implement a tested helper function that performs a deep copy correctly on our data types.

**Comment by AlbertoFabbri93** [comment]
Would this be something similar to `anndata.copy()`?

**Comment by LucaMarconato** [comment]
Yes. It has actually been implemented since a long time: https://github.com/scverse/spatialdata/blob/main/src/spatialdata/_core/_deepcopy.py, we just forgot to close the issue.
